### PR TITLE
Fix "-" in input args not being interpreted as stdin

### LINKED ISSUE #37: Add option to redirect output to stdout instead of modifying file in place
I work with an automatic formatter which needs the file to be outputted to stdout. Expected behaviour:
File `bibliography.bib`:
```
@Book{sweig42,
  Author =	 { Stefa{n} Sweig },
  title =	 { The impossible book },
  publisher =	 { Dead Poet Society},
  year =	 1942,
  month =        mar
}
```

```
bibtex-tidy --quiet --no-in-place bibliography.bib
```
Output:
```
@book{sweig42,
	title        = {The impossible book},
	author       = {Stefa{n} Sweig},
	year         = 1942,
	month        = mar,
	publisher    = {Dead Poet Society}
}
```
Or in many other tools you can do something like this `cat test.py | black --quiet -`, which reads the file from stdin and outputs the formatted file to stdout.

### PR DESCRIPTION
**Pull request by mildblimp**
This fixes #37. Note that I am not that experienced with javascript so feel free to edit it or implement it differently. I did not run tests or add any, sorry!

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import tidy from './index';
-import { readFileSync, writeFileSync } from 'fs';
+import {dash} from 'rw';
+const {readFileSync, writeFileSync} = dash;
```
Comment (FlamingTempura) [inline src/cli.ts:3]: Could you move the const below the imports

**Review by FlamingTempura — CHANGES_REQUESTED** [review]
Thanks for this contribution! Great idea making `-` signal to use stdin 🎉

I got the impression that #37 pertained to stdout not stdin. Does this PR solve that issue?

We aim to have 0 dependencies in bibtex-tidy because it keeps maintenance down, though `rw` is actually quite a simple library so I'd be happy to merge it and try to factor out the important bits later.

This is lacking a test, so can I confirm you've run this and it works?

A few requests before I can merge:
* Add a test, or at least a TODO so someone will remember to add it later.
* Add the types package (`npm i --save-dev @types/rw`). The CI is failing because it's missing.
* Make the indentation and spacing consistent ([prettier](https://prettier.io/) will do this for you).

File: src/cli.ts (hunk 1 above)
Comment (mildblimp) [reply src/cli.ts]: Yes, solved!

File: src/cliUtils.ts [hunk 2]
```diff
@@ -63,7 +63,9 @@ export function splitCLIArgs(
 		// An option is any arg beginning dash or double dash. However, some values
 		// can being with "-", e.g when setting a field to sort in descending order.
 		// When those are encountered, isValue will be true.
-		if (!isValue && arg.startsWith('-')) {
+    // If, on the other hand, the argument is equal to "-", then it is an alias
+    // for stdin and should be processed as a filename. Therefore we skip it.
+    if (!isValue && arg.startsWith('-') && arg != '-') {
```
[the remark this answers is not in the document]
Comment (mildblimp) [reply src/cliUtils.ts]: Solved the indentation

**Comment by mildblimp** [comment]
Thank you for the feedback! I submitted an update.

A few comments:
- #37 pertains to both stdin and stdout, essentially the desired behaviour is that `cat mybib.bib | bibtex-tidy --quiet -`  reads from stdin and outputs to stdout.
- I added a template for a test. However, I don't know how to write the test exactly so I added a todo note as you suggested.
- I used prettier to format the files,although it seemed to also touch parts of the code I didn't write. Maybe your config is different?
- There is no `@types/rw` module, so I added a declaration file.

**Review by FlamingTempura — APPROVED** [review]
Apologies for delay in reviewing, looks great! Thank you for your contribution, I'll merge this in.

**Comment by FlamingTempura** [comment]
Here's the unit test if you're interested: https://github.com/FlamingTempura/bibtex-tidy/commit/4c148107e61b85387970d3455defba8567d2c911#diff-7aea7d757f1f0488ad60917d89606008fdea0e26dead8b8a20f37c67b0b1abcc

You were on the right lines. To get your test working I needed to change the test harness to support stdin. Works nicely 😀

**Pull request merged**